code/_onclick/item_attack.dm

```diff
@@ -149,7 +149,7 @@
 // Proximity_flag is 1 if this afterattack was called on something adjacent, in your square, or on your person.
 // Click parameters is the params string from byond Click() code, see that documentation.
 /obj/item/proc/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
-	SEND_SIGNAL(src, COMSIG_ITEM_AFTERATTACK, target, user, proximity_flag, click_parameters)
+	SEND_SIGNAL(target, COMSIG_ITEM_AFTERATTACK, src, user, target, proximity_flag, click_parameters)
```

Comment:
I'm still not entirely sure if changing this was necessary, I'll have to look into it